src/HDInsight/HDInsight/Models/Management/AzureHdInsightManagementClient.cs

```diff
@@ -225,5 +226,21 @@ public virtual void RestartHosts(string resourceGroupName, string clusterName, I
         {
             HdInsightManagementClient.VirtualMachines.RestartHosts(resourceGroupName, clusterName, hosts);
         }
+        
+        private void ResetClusterIdentity(ClusterCreateParametersExtended createParams, string cloudAadAuthority, string cloudDataLakeAudience)
+        {
+            var configuation = (Dictionary<string, Dictionary<string, string>>)createParams.Properties.ClusterDefinition.Configurations;
+            Dictionary<string, string> clusterIdentity;
+            if(!configuation.TryGetValue("clusterIdentity", out clusterIdentity))
+            {
+                return;
+            }
+            clusterIdentity.SetProperty("clusterIdentity.resourceUri", cloudDataLakeAudience);
```

Comment:
Just throw a question out of curiosity. Is there such a scenario that `clusterIdentity` is not null but `cloudAadAuthority` /`cloudDataLakeAudience` is null?  I am not familiar with  `clusterIdentity`, so if my question is wrong, please correct it.

Reply:
> Just throw a question out of curiosity. Is there such a scenario that `clusterIdentity` is not null but `cloudAadAuthority` /`cloudDataLakeAudience` is null? I am not familiar with `clusterIdentity`, so if my question is wrong, please correct it.

Yes, `cloudDataLakeAudience` may be null. A dictionary can accept a null as value. From the code side, it is ok.